project/SparkCassandraConnectorBuild.scala

```diff
@@ -59,14 +61,82 @@ object CassandraSparkBuild extends Build {
         "org.scala-lang" % "scala-compiler" % scalaVersion.value))
   ).disablePlugins(AssemblyPlugin, SparkPackagePlugin) configs IntegrationTest
 
-  lazy val connector = CrossScalaVersionsProject(
+  /**
+    * Do not included shaded dependencies so they will not be listed in the Pom created for this
+    * project.
+    *
+    * Run the compile from the shaded project since we are no longer including shaded libs
+    */
+  lazy val connectorDistribution = CrossScalaVersionsProject(
     name = namespace,
-    conf = assembledSettings ++ Seq(libraryDependencies ++= Dependencies.connector ++ Seq(
+    conf = assembledSettings ++ Seq(
+      libraryDependencies ++= Dependencies.connectorDistribution ++ Seq(
         "org.scala-lang" % "scala-reflect"  % scalaVersion.value,
-        "org.scala-lang" % "scala-compiler" % scalaVersion.value % "test,it")) ++ pureCassandraSettings
-    ).copy(dependencies = Seq(embedded % "test->test;it->it,test;")
+        "org.scala-lang" % "scala-compiler" % scalaVersion.value % "test,it"),
+      assembly in spPackage := (assembly in shadedConnector).value,
+      //Use the assembly which contains all of the libs not just the shaded ones
+      assembly := (assembly in fullConnector).value,
+      //Use the shaded jar as our packageTarget
+      packageBin := {
+        val shaded = (assembly in shadedConnector).value
+        val targetName = target.value
+        val expected = target.value / s"$namespace-${version.value}.jar"
+        IO.move(shaded, expected)
+        val log = streams.value.log
+        log.info(s"""Shaded jar moved to $expected""".stripMargin)
+        expected
+      },
+      sbt.Keys.`package` := packageBin.value)
+      ++ pureCassandraSettings
+      //Update the distribution tasks to use the shaded jar
+      ++ {for (taskKey <- Seq(publishLocal in Compile, publish in Compile, publishM2 in Compile)) yield {
+      packagedArtifacts in taskKey := {
+        val previous = (packagedArtifacts in Compile).value
+        val shadedJar = (artifact.value.copy(configurations = Seq(Compile)) -> packageBin.value)
+        //Clobber the old build artifact with the shaded jar
+        previous + shadedJar
+      }
+    }}
+  ).copy(dependencies = Seq(embedded % "test->test;it->it,test;")
   ) configs IntegrationTest
 
+  /** Because the distribution project has to mark the shaded jars as provided to remove them from
+    * the distribution dependencies we provide this additional project to build a fat jar which
+    * includes everything. The artifact produced by this project is unshaded while the assembly
```

Comment:
fullConnector `package` will not have shaded guava because it doesn't run through the "assembly" task which is required to actually do the shading. So the trick in the casssandraDistribution is the actual `packageBin` is basically reassigned to the `assembly` task from the shadedJar project. If you take the jar from the package command and unzip it you'll see it doesn't contain a "shaded" directory